SmartEVSE-3/src/evse.cpp

```diff
@@ -3383,34 +3672,21 @@ void StartwebServer(void) {
                     ToModemWaitStateTimer = 0;
                     ToModemDoneStateTimer = 0;
                     LeaveModemDoneStateTimer = 0;
-                    setAccess(0);
+                    setAccess(false);
                     break;
                 case 1:
-                    setAccess(!DelayedStartTime.epoch2);              //if DelayedStartTime not zero then we are Delayed Charging
                     setMode(MODE_NORMAL);
                     break;
                 case 2:
-                    if (MainsMeter) {
-                        OverrideCurrent = 0;
-                        setAccess(!DelayedStartTime.epoch2);
-                        setMode(MODE_SOLAR);
-                        break;
-                    }
+                    setMode(MODE_SOLAR);
+                    break;
                 case 3:
-                    if (MainsMeter) {
-                        setAccess(!DelayedStartTime.epoch2);
-                        setMode(MODE_SMART);
-                        break;
-                    }
+                    setMode(MODE_SMART);
+                    break;
```

Comment:
These two lines seem incorrect to me: if you have no mainsmeter installed, you can not run in Smart mode!

Reply:
Same as with SOLAR.